add FORCE_INVERTED_INDEX hint (#18575)
cockroachdb/docs: src/current/_includes/v24.1/misc/force-index-selection.md

##### Force index scan

To force a scan of a specific index:

{% include_cached copy-clipboard.html %}
~~~ sql
SELECT * FROM table@my_idx;
~~~

This is equivalent to the longer expression:

{% include_cached copy-clipboard.html %}
~~~ sql
SELECT * FROM table@{FORCE_INDEX=my_idx};
~~~

##### Force reverse scan

To force a reverse scan of a specific index:

{% include_cached copy-clipboard.html %}
~~~ sql
SELECT * FROM table@{FORCE_INDEX=my_idx,DESC};
~~~

> Forcing a reverse scan is sometimes useful during [performance tuning]({% link {{ page.version.version }}/performance-best-practices-overview.md %}). For reference, the full syntax for choosing an index and its scan direction is

Reviewer: ```suggestion
Forcing a reverse scan can help with [performance tuning]({% link {{ page.version.version }}/performance-best-practices-overview.md %}). To choose an index and its scan direction:
```

(2 rows)
~~~

#### Force inverted index scan

> To force a scan of any [inverted index]({% link {{ page.version.version }}/inverted-indexes.md %}) of the hinted table:

Reviewer: ```suggestion
To force a scan of an [inverted index]({% link {{ page.version.version }}/inverted-indexes.md %}) of the hinted table:
```

{% include_cached copy-clipboard.html %}
~~~ sql
SELECT * FROM table@{FORCE_INVERTED_INDEX};
~~~

> The `FORCE_INVERTED_INDEX` hint does not allow specifying an inverted index. If no query plan can be generated, the query will result in the error:

Reviewer: I don't understand this sentence, but maybe it's just because I'm not familiar with the optimizer or indexes. I would expect the session variable to do exactly what it says it doesn't do.

Author: Yeah, it's because unlike the other hints, this one can't be appended with a specific index name to scan. I think that should be clear on this page, since the other examples show this.
